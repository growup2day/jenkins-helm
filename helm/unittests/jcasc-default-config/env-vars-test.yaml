templates:
  - config-jcasc.yaml
tests:
  - it: should include environment variables in default config script when configured
    set:
      JCasC:
        defaultConfig: true
        envVars:
          VAR_NAME_1: "var value 1"
          var_2: "var value 2"
    asserts:
      - isNotNull:
          path: data.[jcasc-default-config-env-vars.yaml]
      - matchSnapshot:
          path: data.[jcasc-default-config-env-vars.yaml]
  - it: should include blank env-vars config in default config scripts when not configured
    set:
      JCasC:
        defaultConfig: true
    asserts:
      - isNotNull:
          path: data.[jcasc-default-config-env-vars.yaml]
      - matchSnapshot:
          path: data.[jcasc-default-config-env-vars.yaml]
  - it: should not include env-vars config in default config scripts when there is custom config script for the section
    set:
      JCasC:
        defaultConfig: true
        configScripts:
          custom-override: |
            jenkins:
              globalNodeProperties:
                - envVars:
                    env: []
    asserts:
      - isNull:
          path: data.[jcasc-default-config-env-vars.yaml]
      - isNotNull:
          path: data.[custom-override.yaml]
