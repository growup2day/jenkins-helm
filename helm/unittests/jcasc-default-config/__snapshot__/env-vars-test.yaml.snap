should include blank env-vars config in default config scripts when not configured:
  1: |
    |-
      jenkins:
        # Configure System > Global properties
        globalNodeProperties:
          - envVars:
              env: []
should include environment variables in default config script when configured:
  1: |
    |-
      jenkins:
        # Configure System > Global properties
        globalNodeProperties:
          - envVars:
              env:
                - key: "VAR_NAME_1"
                  value: "var value 1"
                - key: "var_2"
                  value: "var value 2"
